provider/oci/instance.go

```diff
@@ -4,44 +4,284 @@
 package oci
 
 import (
+	"context"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
 	"github.com/juju/errors"
+	"github.com/juju/juju/status"
 
-	"github.com/juju/juju/environs/context"
+	envcontext "github.com/juju/juju/environs/context"
+	"github.com/juju/juju/environs/instances"
 	"github.com/juju/juju/instance"
 	"github.com/juju/juju/network"
+
+	ociCore "github.com/oracle/oci-go-sdk/core"
+)
+
+const (
+	// MinVolumeSizeMB is the minimum size in MB for a volume or boot disk
+	MinVolumeSizeMB = 51200
+
+	// MaxVolumeSizeMB is the maximum size in MB for a volume or boot disk
+	MaxVolumeSizeMB = 16777216
 )
 
-type ociInstance struct{}
+type ociInstance struct {
+	arch     string
+	instType *instances.InstanceType
+	env      *Environ
+	mutex    *sync.Mutex
```

Comment:
Any reason not to just use ```sync.Mutex``` and remove the assignment in ```newInstance```?

Reply:
nope, changed!